2023-01-06 11:10:32   Arguments: Namespace(aggregation='gem', backbone='resnet18conv4', brightness=0, cache_refresh_rate=1000, contrast=0, criterion='triplet', dataset_name='pitts30k', datasets_folder='/content/drive/MyDrive/MLDL/datasets_vg', device='cuda', efficient_ram_testing=False, epochs_num=10, fc_output_dim=None, freeze_te=None, horizontal_flip=False, hue=0, infer_batch_size=16, l2='before_pool', lr=1e-05, lr_crn_layer=0.005, lr_crn_net=0.0005, majority_weight=0.01, margin=0.1, mining='partial', neg_samples_num=1000, negs_num_per_query=10, netvlad_clusters=64, num_workers=2, off_the_shelf='imagenet', optim='adam', patience=3, pca_dataset_folder=None, pca_dim=None, pretrain='imagenet', queries_per_epoch=5000, rand_perspective=0, random_resized_crop=0, random_rotation=0, recall_values=[1, 5, 10, 20], resize=[480, 640], resume=None, saturation=0, save_dir='/content/drive/MyDrive/MLDL/results/base_gem/2023-01-06_11-10-32', seed=0, test_method='hard_resize', train_batch_size=4, train_positives_dist_threshold=10, trunc_te=None, val_positive_dist_threshold=25)
2023-01-06 11:10:32   The outputs are being saved in /content/drive/MyDrive/MLDL/results/base_gem/2023-01-06_11-10-32
2023-01-06 11:10:32   Using 1 GPUs and 2 CPUs
2023-01-06 11:10:42   There are 96 queries without any positives within the training set. They won't be considered as they're useless for training.
2023-01-06 11:10:42   Train query set: < TripletsDataset, pitts30k - #database: 10000; #queries: 7320 >
2023-01-06 11:12:40   Val set: < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >
2023-01-06 11:14:33   Test set: < BaseDataset, pitts30k - #database: 10000; #queries: 6816 >
2023-01-06 11:14:49   Output dimension of the model is 256, with 17.29 GFLOPs
2023-01-06 11:14:49   Start training epoch: 00
2023-01-06 12:42:15   Finished epoch 00 in 1:27:26, average epoch triplet loss = 0.0524
2023-01-06 14:33:19   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 72.0, R@5: 87.1, R@10: 91.6, R@20: 94.7
2023-01-06 14:33:19   Improved: previous best R@5 = 0.0, current R@5 = 87.1
2023-01-06 14:33:19   Start training epoch: 01
2023-01-06 15:19:28   Finished epoch 01 in 0:46:08, average epoch triplet loss = 0.0338
2023-01-06 15:23:40   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 75.4, R@5: 89.1, R@10: 92.8, R@20: 95.6
2023-01-06 15:23:40   Improved: previous best R@5 = 87.1, current R@5 = 89.1
2023-01-06 15:23:40   Start training epoch: 02
2023-01-06 15:58:17   Finished epoch 02 in 0:34:36, average epoch triplet loss = 0.0245
2023-01-06 16:02:28   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 76.6, R@5: 89.9, R@10: 93.2, R@20: 96.0
2023-01-06 16:02:28   Improved: previous best R@5 = 89.1, current R@5 = 89.9
2023-01-06 16:02:28   Start training epoch: 03
2023-01-06 16:31:25   Finished epoch 03 in 0:28:57, average epoch triplet loss = 0.0191
2023-01-06 16:35:38   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 77.1, R@5: 90.5, R@10: 93.7, R@20: 96.3
2023-01-06 16:35:38   Improved: previous best R@5 = 89.9, current R@5 = 90.5
2023-01-06 16:35:38   Start training epoch: 04
2023-01-06 17:01:27   Finished epoch 04 in 0:25:48, average epoch triplet loss = 0.0152
2023-01-06 17:05:51   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 77.3, R@5: 90.6, R@10: 93.9, R@20: 96.4
2023-01-06 17:05:51   Improved: previous best R@5 = 90.5, current R@5 = 90.6
2023-01-06 17:05:51   Start training epoch: 05
2023-01-06 17:29:54   Finished epoch 05 in 0:24:02, average epoch triplet loss = 0.0117
2023-01-06 17:34:10   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 77.9, R@5: 91.1, R@10: 94.1, R@20: 96.4
2023-01-06 17:34:10   Improved: previous best R@5 = 90.6, current R@5 = 91.1
2023-01-06 17:34:10   Start training epoch: 06
2023-01-06 17:57:16   Finished epoch 06 in 0:23:05, average epoch triplet loss = 0.0101
2023-01-06 18:01:36   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 77.4, R@5: 90.7, R@10: 93.9, R@20: 96.5
2023-01-06 18:01:37   Not improved: 1 / 3: best R@5 = 91.1, current R@5 = 90.7
2023-01-06 18:01:37   Start training epoch: 07
2023-01-06 18:24:08   Finished epoch 07 in 0:22:31, average epoch triplet loss = 0.0086
2023-01-06 18:28:25   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 77.9, R@5: 91.2, R@10: 94.4, R@20: 96.7
2023-01-06 18:28:25   Improved: previous best R@5 = 91.1, current R@5 = 91.2
2023-01-06 18:28:25   Start training epoch: 08
2023-01-06 18:50:48   Finished epoch 08 in 0:22:22, average epoch triplet loss = 0.0073
2023-01-06 18:55:05   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 78.2, R@5: 91.2, R@10: 94.2, R@20: 96.5
2023-01-06 18:55:05   Not improved: 1 / 3: best R@5 = 91.2, current R@5 = 91.2
2023-01-06 18:55:05   Start training epoch: 09
2023-01-06 19:17:20   Finished epoch 09 in 0:22:14, average epoch triplet loss = 0.0058
2023-01-06 19:21:37   Recalls on val set < BaseDataset, pitts30k - #database: 10000; #queries: 7608 >: R@1: 78.4, R@5: 90.9, R@10: 94.2, R@20: 96.5
2023-01-06 19:21:37   Not improved: 2 / 3: best R@5 = 91.2, current R@5 = 90.9
2023-01-06 19:21:37   Best R@5: 91.2
2023-01-06 19:21:37   Trained for 10 epochs, in total in 8:11:04
2023-01-06 19:54:45   
Traceback (most recent call last):
  File "train.py", line 216, in <module>
    recalls, recalls_str = test.test(args, test_ds, model, test_method=args.test_method)
  File "/content/drive/MyDrive/MLDL/deep-visual-geo-localization-benchmark-main/test.py", line 144, in test
  File "/usr/local/lib/python3.8/dist-packages/tqdm/std.py", line 1195, in __iter__
    for obj in iterable:
  File "/usr/local/lib/python3.8/dist-packages/torch/utils/data/dataloader.py", line 628, in __next__
    data = self._next_data()
  File "/usr/local/lib/python3.8/dist-packages/torch/utils/data/dataloader.py", line 1333, in _next_data
    return self._process_data(data)
  File "/usr/local/lib/python3.8/dist-packages/torch/utils/data/dataloader.py", line 1359, in _process_data
    data.reraise()
  File "/usr/local/lib/python3.8/dist-packages/torch/_utils.py", line 543, in reraise
    raise exception
FileNotFoundError: Caught FileNotFoundError in DataLoader worker process 1.
Original Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/torch/utils/data/_utils/worker.py", line 302, in _worker_loop
    data = fetcher.fetch(index)
  File "/usr/local/lib/python3.8/dist-packages/torch/utils/data/_utils/fetch.py", line 58, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.8/dist-packages/torch/utils/data/_utils/fetch.py", line 58, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.8/dist-packages/torch/utils/data/dataset.py", line 295, in __getitem__
    return self.dataset[self.indices[idx]]
  File "/content/drive/MyDrive/MLDL/deep-visual-geo-localization-benchmark-main/datasets_ws.py", line 106, in __getitem__
  File "/content/drive/MyDrive/MLDL/deep-visual-geo-localization-benchmark-main/datasets_ws.py", line 25, in path_to_pil_img
  File "/usr/local/lib/python3.8/dist-packages/PIL/Image.py", line 2843, in open
    fp = builtins.open(filename, "rb")
FileNotFoundError: [Errno 2] No such file or directory: '/content/drive/MyDrive/MLDL/datasets_vg/pitts30k/images/test/database/@0584553.43@4476945.19@17@T@040.43885@-080.00304@002709@32@@@@@@pitch2_yaw9@.jpg'


